app/src/server/tasks/evaluateTestSetEntry.task.ts

```diff
@@ -9,48 +10,58 @@ import {
   getStringsToPrune,
 } from "~/modelProviders/fine-tuned/getCompletion";
 import { countLlamaChatTokens, countLlamaChatTokensInMessages } from "~/utils/countTokens";
-import { getCompletion2 } from "~/modelProviders/fine-tuned/getCompletion-2";
+import { getCompletion2, getOpenaiCompletion } from "~/modelProviders/fine-tuned/getCompletion-2";
 import { calculateEntryScore } from "../utils/calculateEntryScore";
 import { typedDatasetEntry } from "~/types/dbColumns.types";
+import { getComparisonModelName, isComparisonModel } from "~/utils/baseModels";
 
 export type EvaluateTestSetEntryJob = {
-  fineTuneId: string;
+  modelId: string;
   datasetEntryId: string;
   skipCache?: boolean;
 };
 
 export const evaluateTestSetEntry = defineTask<EvaluateTestSetEntryJob>({
   id: "evaluateTestSetEntry",
   handler: async (task) => {
-    const { fineTuneId, datasetEntryId, skipCache } = task;
+    const { modelId, datasetEntryId, skipCache } = task;
 
-    const [fineTune, rawDatasetEntry] = await prisma.$transaction([
-      prisma.fineTune.findUnique({
-        where: { id: fineTuneId },
-      }),
-      prisma.datasetEntry.findUnique({
-        where: { id: datasetEntryId },
-      }),
-    ]);
+    const rawDatasetEntry = await prisma.datasetEntry.findUnique({
+      where: { id: datasetEntryId },
+      include: {
+        dataset: {
+          include: {
+            project: true,
+          },
+        },
+      },
+    });
 
-    if (!fineTune || !rawDatasetEntry) return;
+    if (!rawDatasetEntry?.dataset) return;
 
-    const datasetEntry = typedDatasetEntry(rawDatasetEntry);
+    let fineTune;
+    if (!isComparisonModel(modelId)) {
+      fineTune = await prisma.fineTune.findUnique({
+        where: { id: modelId },
+      });
+    }
+
+    const datasetEntry = typedDatasetEntry(omit(rawDatasetEntry, "dataset"));
 
     const existingTestEntry = await prisma.fineTuneTestingEntry.findUnique({
-      where: { fineTuneId_datasetEntryId: { fineTuneId, datasetEntryId } },
+      where: { modelId_datasetEntryId: { modelId, datasetEntryId } },
     });
 
     if (existingTestEntry?.output && !skipCache) return;
 
     let prunedMessages = existingTestEntry?.prunedInput;
 
-    const stringsToPrune = await getStringsToPrune(fineTune.id);
+    const stringsToPrune = await getStringsToPrune(modelId);
```

Comment:
Maybe we can just drop pruned messages from the test set entries? Not sure we actually need them there?

Reply:
We definitely do need to prune messages for the test set entries, because some fine-tunes have been trained on pruned messages.

Reply:
We need them when we run inference yes. But `getCompletion2` handles that internally. It looks like we aren't using any of `prunedInput`, `prunedInputTokens`, `outputTokens` on the `FineTuneTestingEntry` and could just drop them?

Reply:
As we discussed, we'll want to start showing cost soon, so we need prunedInputTokens and outputTokens. I think keeping prunedInput also makes sense (though I suppose we could calculate it on the fly) because if the test output is horribly wrong, it would be useful to see what the pruned input looked like.

Reply:
Yeah let's drop `prunedInput` from the db and calculate it on the fly if necessary.